Uploading the CSVReader program that reads from a csv file and prints…

kschetan92 opened the pull request:
Uploading the CSVReader program that reads from a Employee.csv file and prints the results based on search condition, Also can create new employees.

kschetan92 committed:
Uploading the CSVReader program that reads from a csv file and prints the results based on search condition

manoj-v2 commented:
What happens if i create another class named students and extend CSVReader. They will inherit name, age and department.  Also what if I go ahead and change the Employee.csv and add more columns ?

kschetan92 committed:
Modified the program for reading different csv files and different subclasses

kschetan92 commented:
Have tried to fix reading of different files. Now working on reading the headers dynamically from the CSV files to send to attr_accessor.

manoj-v2 commented:
Create multiple CSV files and test your program. Dont just rely on one csv file only.

kschetan92 committed:
Modifying the program to read different files and accessing method dynamically

kschetan92 committed:
Modifying the program to read different files and accessing method dynamically

kschetan92 committed:
Modifying the program where the find_by_ methods are attached with respect to the Subclasses only

kschetan92 commented:
Added the CSV files to the directory, Modified it to stick/attach the find_by_methods with respect to subclasses only. 

Thank you for the guidance sir.

manoj-v2 commented on CSVReader/CSVReaders.rb:
```diff
@@ -0,0 +1,42 @@
+          subclass_object = #{@subclass}.new 
+          p subclass_object
+          p subclass_object = row
+        end
```
This method is getting created more than once. The method name is same. So it should be moved out of the  @headers.each loop.

manoj-v2 replied:
Also when you do a find_by_xxx  it does return an object. But are you sure that  attributes (columns names based accessors) have those values.. 

emp = Employee.find_by_department("Finance") 
p emp.name
p emp.age
p emp.department

These attributes should return you the matched result

kschetan92 committed:
Moving the process_results method out of the @headers.each loop

kschetan92 committed:
Modifying the program to print matching output and also print specific data from the match

kschetan92 commented:
the program is modified to print the matching data as required

manojpm commented on CSVReader/CSVReaders.rb:
```diff
@@ -0,0 +1,55 @@
+  end
+
+    class_eval %Q"
+        def #{@subclass}.process_results(row)
```
This method needs to be out of the loop.

kschetan92 committed:
Enhancing the program with adding Exceptions

kschetan92 committed:
Enhancing the program with adding Exceptions

kschetan92 commented:
I have tried to add some Exceptions to the program, that would raise the custom message if the file is not found in the path or if the file is not found at all.
